apps/condo/domains/organization/components/DeleteEmployeeButtonWithReassignmentModel.tsx

```diff
@@ -0,0 +1,248 @@
+
+    const [newEmployeeUserId, setNewEmployeeUserId] = useState(null)
+    const onChange = (newEmployeeUserId: string) => {
+        setNewEmployeeUserId(newEmployeeUserId)
+    }
+
+    const [isDeleting, setIsDeleting] = useState(false)
+    const [isConfirmVisible, setIsConfirmVisible] = useState(false)
+
+    const showConfirm = () => setIsConfirmVisible(true)
+    const handleCancel = () => setIsConfirmVisible(false)
+    
+    const [loadTicketsToReassign, { error: errorLoadTickets }] = useGetOrganizationEmployeeTicketsForReassignmentLazyQuery({
+        variables: {
+            organizationId: employeeOrganizationId,
+            userId: employeeUserId,
+            first: 100,
+        },
+        fetchPolicy: 'no-cache',
+    })
+    const [updateTickets, { error: errorUpdateTickets }] = useUpdateOrganizationEmployeeTicketsForReassignmentMutation()
+
+    const handleDeleteButtonClick = () => {
+        setIsDeleting(true)
+        setIsConfirmVisible(false)
+
+        runMutation(
+            {
+                action: softDeleteAction,
+                onError: (e) => { throw e },
+                OnErrorMsg: () => (getNotificationInfo(ERROR_NOTIFICATION_TYPE)),
+                onFinally: () => setIsDeleting(false),
+                intl,
+            },
+        )
+    }
+    
+    const handleDeleteAndReassignTicketsClick = async () => {
+        setIsDeleting(true)
+        setIsConfirmVisible(false)
+
+        let updatedTicketsCount = 0
+
+        /* NOTE: push notifications for bulk tickets updates should not be sent here */
+        while (updatedTicketsCount < activeTicketsOrganizationEmployeeCount) {
+            notificationApi.warning(getNotificationInfo(WARNING_NOTIFICATION_TYPE, updatedTicketsCount))
+
+            try {
+                const { data: ticketsToReassign } = await loadTicketsToReassign()
+
+                if (isEmpty(ticketsToReassign?.tickets)) break
+                const { data: reassignedTickets }  = await updateTickets({
+                    variables: {
+                        data: ticketsToReassign?.tickets?.filter(Boolean).map(ticket => ({
+                            id: ticket?.id,
+                            data: getTicketReassignData(ticket),
+                        })),
+                    },
+                })
+                
+                updatedTicketsCount += reassignedTickets?.tickets?.length
+                await waitBetweenRequsted()
+            } catch (err) {
+                if (errorLoadTickets || errorUpdateTickets || err) notificationApi.error(getNotificationInfo(ERROR_NOTIFICATION_TYPE))
+                setIsDeleting(false)
+                return
+            }
+        }
+
+        if (activeTicketsOrganizationEmployeeCount === updatedTicketsCount) {
```

Comment:
do we really need this condition? If the number of tickets for an employee has changed up or down (not because of our feature), then an error will return, do we need this behavior?